tests/utils/test_modeling_tf_core.py

```diff
@@ -218,6 +218,10 @@ def test_saved_model_creation_extended(self):
             model = model_class(config)
             num_out = len(model(class_inputs_dict))
 
+            for key in list(class_inputs_dict.keys()):
+                if isinstance(class_inputs_dict[key], tf.Tensor) and class_inputs_dict[key].dtype.is_integer:
```

Comment:
This feels like a bit of a hack - shouldn't `self._prepare_for_class` return a dict with the correct types?

Reply:
I thought some model classes might put a bool in there! Maybe I'm wrong, though

Reply:
Will do!